src/common/syncjournaldb.cpp

```diff
@@ -64,6 +66,7 @@ static void fillFileRecordFromGetQuery(SyncJournalFileRecord &rec, SqlQuery &que
     rec._fileSize = query.int64Value(7);
     rec._serverHasIgnoredFiles = (query.intValue(8) > 0);
     rec._checksumHeader = query.baValue(9);
+    rec._hasDirtyPlaceholder = query.intValue(10);
```

Comment:
Further down the `hasDirtyPlaceholder` column is defined as `BOOLEAN` - is converting it to int here the best approach, or is there a more natural `toBool`?

Reply:
Your implementation treats integers and booleans the same :D
https://github.com/owncloud/client/blob/a3e8fa7dad7e29d0cb2ef76bc49f35c18f03e00e/src/common/ownsql.cpp#L412-L415

Reply:
Well, that is a decade old at least... maybe that is not optimal as well?